ecmd-core/cmd/ecmdRingUser.C

```diff
@@ -3285,19 +3285,18 @@ uint32_t readScandefFile(ecmdChipTarget & target, const char* i_ringName, ecmdDa
   uint32_t ringBeginOffset = 0;
   uint32_t numRings =0;
 
-  rc = ecmdQueryFileLocation(target, ECMD_FILE_SCANDEF, l_filePairs, l_version);
+  rc = ecmdQueryFileLocation(target, ECMD_FILE_SCANDEF, l_fileLocs, l_version);
   if (rc)
   {
       printed = "readScandefFile - Error occured locating scandef file\n";
       ecmdOutputError(printed.c_str());
       return rc;
   }
 
-  std::list<std::pair<std::string, std::string> >::iterator l_filePair = l_filePairs.begin();
-  while (l_filePair != l_filePairs.end())
+  for (auto l_fileLoc : l_fileLocs)
```

Comment:
This is going to cause problems with some of the old 32 bit compilers for architectures we support.